frontend/controller/actions/utils.js

```diff
@@ -313,3 +305,48 @@ export async function createInvite ({ quantity = 1, creatorID, expires, invitee
     invitee
   }
 }
+
+export function groupContractsByType (contracts: Object): Object {
+  const contractIDs = Object.create(null)
+  if (contracts) {
+    // $FlowFixMe[incompatible-use]
+    Object.entries(contracts).forEach(([id, { type }]) => {
+      if (!contractIDs[type]) {
+        contractIDs[type] = []
+      }
+      contractIDs[type].push(id)
+    })
+  }
+  return contractIDs
+}
+
+export async function syncContractsInOrder (identityContractID: string | null, queuePrefix: string, groupedContractIDs: Object): Promise<any> {
```

Comment:
This `queuePrefix` seems entirely unnecessary. Can it be removed?

Reply:
Why is it unnecessary? It's used to select the queue name, as this can be called when starting the app or when logging in (with a password).

Reply:
I see no reason for it to be called on an event queue. What is this protecting from?

`'chelonia/contract/sync'` already uses an event queue.

Reply:
sync might use an event queue, but this is more about initialisation. In particular, the `login:` queue is used to prevent multiple login (logout) events being handled out of order. It's probably not that important for the app init phase though.

Reply:
How can there be multiple login events at the same time?

Reply:
sbp('gi.app/identity/login', ...)
sbp('gi.app/identity/login', ...)
sbp('gi.app/identity/login', ...)

Reply:
1. That does not answer my question, which is about how that can happen.
2. It's unclear to me why there are two separate queues for this, `appStart` and `login`, when you could just have one queue inside of `syncContractsInOrder` (no need for parameter)
3. It's not clear why even a single queue would be needed, since `'chelonia/contract/sync'` already uses a queue